### PR TITLE
 HIVE-27797 - ADDENDUM Transactions that got timed out are not gettin…

### PR DESCRIPTION
**Pull request by tarak271**
…g logged as ABORTED - Flaky test case

### What changes were proposed in this pull request?

Correcting a flaky test by cleaning Notifications after every run which are causing failures in subsequent runs

### Why are the changes needed?

correct flaky test case

### Does this PR introduce _any_ user-facing change?

No
### Is the change a dependency upgrade?

No

### How was this patch tested?

Junit test cases

### CODE REVIEW

File: itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestTimedOutTxnNotificationLogging.java [hunk 1]
```diff
@@ -194,6 +191,9 @@ private void runHouseKeeperService() {
   }
 
   private void runTxnHouseKeeperService() {
+    ObjectStore objectStore = new ObjectStore();
```
Comment (deniskuzZ) [inline itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestTimedOutTxnNotificationLogging.java:194]: move ObjectStore initialization in setUp

File: itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestTimedOutTxnNotificationLogging.java [hunk 2]
```diff
@@ -93,7 +91,6 @@ private void setConf() {
     MetastoreConf.setBoolVar(hiveConf, MetastoreConf.ConfVars.HIVE_IN_TEST, true);
     MetastoreConf.setVar(hiveConf, MetastoreConf.ConfVars.WAREHOUSE, "/tmp");
     MetastoreConf.setTimeVar(hiveConf, MetastoreConf.ConfVars.TXN_TIMEOUT, 1000, TimeUnit.MILLISECONDS);
```
Comment (deniskuzZ) [inline itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestTimedOutTxnNotificationLogging.java:93]: could you please use the same scale: TimeUnit.SECONDS

File: itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestTxnNotificationLogging.java [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+    AtomicInteger numTxns = new AtomicInteger();
+    hiveDb.getMSC().showTxns().getOpen_txns().forEach(txnInfo -> {
+      if (txnInfo.getState() == txnState && txnIds.contains(txnInfo.getId())) {
+        numTxns.incrementAndGet();
+      }
+    });
+    return numTxns.get();
+  }
+
+  private void runHouseKeeperService() {
```
Comment (deniskuzZ) [inline itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestTxnNotificationLogging.java:171]: can you create a base class and move there common methods?

File: itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestTimedOutTxnNotificationLogging.java (hunk 1 above)
Comment (tarak271) [reply itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestTimedOutTxnNotificationLogging.java]: Moved them to setup

File: itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestTimedOutTxnNotificationLogging.java (hunk 2 above)
Comment (tarak271) [reply itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestTimedOutTxnNotificationLogging.java]: changed

File: itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestTxnNotificationLogging.java (hunk 3 above)
Comment (tarak271) [reply itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestTxnNotificationLogging.java]: This class is not needed now, removed it

**Review by deniskuzZ — APPROVED** [review]
LGTM +1

**Comment by deniskuzZ** [comment]
hi @tarak271 , could you please rebase

**Comment by tarak271** [comment]
@deniskuzZ rebased as recommended

**Pull request merged**